binaries/cli/src/attach.rs

```diff
@@ -0,0 +1,143 @@
+                            })?;
+                            node_path_lookup
+                                .insert(path, (dataflow_id, node.id.clone(), Some(op.id.clone())));
+                        }
+                        _ => (), // TODO: Reloading for non-Python Operator
+                    }
+                }
+            }
+        }
+    }
+
+    // Setup dataflow file watcher if reload option is set.
+    let watcher_tx = tx.clone();
+    let _watcher = if hot_reload {
+        let hash = node_path_lookup.clone();
+        let paths = hash.keys();
+        let notifier = move |event| match event {
+            Ok(NotifyEvent {
+                paths,
+                kind: EventKind::Modify(ModifyKind::Data(_data)),
+                ..
+            }) => {
+                for path in paths {
+                    let (dataflow_id, node_id, operator_id) =
+                        node_path_lookup.get(&path).unwrap().clone();
+                    watcher_tx
+                        .send(ControlRequest::Reload {
+                            dataflow_id,
+                            node_id,
+                            operator_id,
+                        })
+                        .unwrap();
+                }
+            }
+            _ => (), // TODO: Manage different file event
+        };
+
+        let mut watcher = RecommendedWatcher::new(
+            notifier,
+            Config::default().with_poll_interval(Duration::from_secs(1)),
+        )?;
+
+        for path in paths {
+            watcher.watch(path, RecursiveMode::Recursive)?;
+        }
+        Some(watcher)
+    } else {
+        None
+    };
+
+    // Setup Ctrlc Watcher to stop dataflow after ctrlc
+    let ctrlc_tx = tx.clone();
+    let mut ctrlc_sent = false;
+    ctrlc::set_handler(move || {
+        if ctrlc_sent {
+            std::process::abort();
+        } else {
+            if ctrlc_tx
+                .send(ControlRequest::Stop {
+                    dataflow_uuid: dataflow_id,
+                })
+                .is_err()
+            {
+                // bail!("failed to report ctrl-c event to dora-daemon");
+            }
+            ctrlc_sent = true;
+        }
+    })
+    .wrap_err("failed to set ctrl-c handler")?;
+
+    loop {
+        let control_request = match rx.recv_timeout(Duration::from_secs(1)) {
+            Err(_err) => ControlRequest::Check {
+                dataflow_uuid: dataflow_id,
+            },
+            Ok(reload_event) => reload_event,
+        };
+
+        let reply_raw = control_connection(session)?
+            .request(&serde_json::to_vec(&control_request)?)
+            .wrap_err("failed to send request message to coordinator")?;
+        let result: ControlRequestReply =
+            serde_json::from_slice(&reply_raw).wrap_err("failed to parse reply")?;
+        match result {
+            ControlRequestReply::DataflowStarted { uuid } => info!("dataflow {uuid} started"),
```

Comment:
Do we want to print the line on every check?

Reply:
Nope!